pyfar/constants/speed_of_sound.py

```diff
@@ -52,3 +52,113 @@ def speed_of_sound_simple(temperature):
+    pressure, humidity and CO2 concentration.
+
+    This implements Cramers method described in [#]_.
+    The uncertainty in the speed of sound is similarly estimated to be
+    less than 300 parts per million.
+
+    Parameters
+    ----------
+    temperature : float, array_like
+        Temperature in degree Celsius from 0°C to 30°C.
+    relative_humidity : float, array_like
+        Relative humidity in the range of 0 to 1.
+    atmospheric_pressure : float, array_like, optional
+        Atmospheric pressure in pascal, by default
+        :py:attr:`reference_atmospheric_pressure`.
+        It must be between 75 000 Pa to 102000 Pa.
+    co2_ppm : float, array_like
+        CO2 concentration in parts per million. The default is 314 ppm,
+        based on Table I. It must be at below 10 000 ppm (1%).
+
+    Returns
+    -------
+    speed_of_sound : float, array_like
+        Speed of sound in air in (m/s).
+
+    References
+    ----------
+    .. [#] O. Cramer, “The variation of the specific heat ratio and the
+           speed of sound in air with temperature, pressure, humidity, and CO2
+           concentration,” The Journal of the Acoustical Society of America,
+           vol. 93, no. 5, pp. 2510-2516, May 1993, doi: 10.1121/1.405827.
+    """
+    if atmospheric_pressure is None:
+        atmospheric_pressure = pf.constants.reference_atmospheric_pressure
+    # convert to array
+    temperature = np.array(temperature)
+    relative_humidity = np.array(relative_humidity)
+    atmospheric_pressure = np.array(atmospheric_pressure)
+    co2_ppm = np.array(co2_ppm)
+
+    # check inputs:
+    if np.any(temperature < 0) or np.any(temperature > 30):
+        raise ValueError("Temperature must be between 0°C and 30°C.")
+    if np.any(relative_humidity < 0) or np.any(relative_humidity > 1):
+        raise ValueError("Relative humidity must be between 0 and 1.")
+    if np.any(atmospheric_pressure < 75e3) or np.any(
+            atmospheric_pressure > 102e3):
+        raise ValueError(
+            "Atmospheric pressure must be between 75 000 Pa to 102 000 Pa.")
+    if np.any(co2_ppm < 0) or np.any(co2_ppm > .01e6):
+        raise ValueError(
+            "CO2 concentration (ppm) must be between 0 ppm to 10 000 ppm.")
+
+    # mole fraction of CO2 in air
+    x_c = co2_ppm * 1e-6
+    # pressure(atmospheric pressure) in Pa
+    p = atmospheric_pressure
+    # relative humidity as a fraction
+    h = relative_humidity
+    # temperature in Celsius
+    t = temperature
+    # thermodynamic temperature
+    T = t - pf.constants.absolute_zero_celsius
+
+    # enhancement factor f (Equation A2)
+    f = 1.00062+3.14e-8 * p + 5.6e-7*temperature**2
+    # saturation vapor pressure in air (Equation A3)
+    p_sv = np.exp(1.2811805e-5*T**2-1.9509874e-2*T+34.04926034-6.3536311e3/T)
+    # water vapor mole fraction (Equation A1)
+    x_w = h * f * p_sv / p
+
+    # check water mole fraction
+    # due to the input checks, this is not possible to reach
```

Comment:
What do you mean with this comment?

Reply:
It cannot be tested, uts its mentioned in the paper,